polymath/qube.py

```
@@ -1902,20 +1902,36 @@ def without_unit(self, *, recursive=True):
         obj = self.clone(recursive=recursive)
         obj._unit = None
+
+        # Strip units from derivatives if recursive is True
+        if recursive and obj._derivs:
+            for key, deriv in obj._derivs.items():
+                if deriv._unit is not None:
+                    obj._derivs[key] = deriv.without_unit(recursive=True)
+
         return obj
 
     def into_unit(self, recursive=False):
```

Ruff (0.14.7)

Boolean default positional argument in function definition

(FBT002)